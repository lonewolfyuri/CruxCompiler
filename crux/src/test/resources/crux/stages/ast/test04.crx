// test index operator

array x: int[10];

func main() : void {
  ::printInt(x[2]);
}
