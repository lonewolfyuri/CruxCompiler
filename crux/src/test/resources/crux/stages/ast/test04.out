ast.DeclarationList(3,0)
  ast.ArrayDeclaration(3,0)[Symbol(x)]
  ast.FunctionDefinition(5,0)[Symbol(main), []]
    ast.StatementList(6,2)
      ast.Call(6,2)[Symbol(printInt)]
        ast.Dereference(6,13)
          ast.ArrayAccess(6,15)
            ast.Name(6,13)[x]
            ast.LiteralInt(6,15)[2]
